File: spring-data-requery/src/main/java/org/springframework/data/requery/converters/ZonedDateTimeToStringConverter.java
package org.springframework.data.requery.converters;

import io.requery.Converter;
import org.springframework.util.StringUtils;

import javax.annotation.Nullable;
import java.time.ZonedDateTime;

/**
 * org.springframework.data.requery.converters.ZonedDateTimeToStringConverter
 *
 * @author debop
 */
public class ZonedDateTimeToStringConverter implements Converter<ZonedDateTime, String> {

    @Override
    public Class<ZonedDateTime> getMappedType() {
        return ZonedDateTime.class;
    }

    @Override
    public Class<String> getPersistedType() {
        return String.class;
    }

    @Nullable
    @Override
    public Integer getPersistedSize() {
        return null;
    }

    @Override
    public String convertToPersisted(ZonedDateTime value) {
        return (value != null)
               ? Java8Times.toIsoZonedDateTimeString(value)
               : null;
    }

    @Override
    public ZonedDateTime convertToMapped(Class<? extends ZonedDateTime> type, @Nullable String value) {
        return (StringUtils.hasText(value))
               ? ZonedDateTime.parse(value)
               : null;
    }
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/converters/ZonedDateTimeToStringConverter.kt
package org.springframework.data.requery.converters

import io.requery.Converter
import java.time.ZonedDateTime

/**
 * Convert [ZonedDateTime] to [String]
 *
 * @author debop
 * @since 18. 7. 2
 */
class ZonedDateTimeToStringConverter: Converter<ZonedDateTime, String> {

    override fun getPersistedType(): Class<String> = String::class.java
    override fun getMappedType(): Class<ZonedDateTime> = ZonedDateTime::class.java
    override fun getPersistedSize(): Int? = null

    override fun convertToMapped(type: Class<out ZonedDateTime>?, value: String?): ZonedDateTime? {
        return value?.let { ZonedDateTime.parse(it) }
    }

    override fun convertToPersisted(value: ZonedDateTime?): String? {
        return value?.toIsoZonedDateTimeString()
    }

}